PWGHF/DataModel/CandidateReconstructionTables.h

```diff
@@ -1531,6 +1531,179 @@ DECLARE_SOA_TABLE(HfCandScMcGen, "AOD", "HFCANDSCMCGEN", //!
                   hf_cand_sigmac::FlagMcMatchGen,
                   hf_cand_sigmac::OriginMcGen);
 
+/// D*(+) --> D0 + π(+)
+namespace hf_cand_dstar
+{
+DECLARE_SOA_EXPRESSION_COLUMN(D0Px, d0px, float, 1.f * aod::hf_cand::pxProng0 + 1.f * aod::hf_cand::pxProng1);
+DECLARE_SOA_EXPRESSION_COLUMN(D0Py, d0py, float, 1.f * aod::hf_cand::pyProng0 + 1.f * aod::hf_cand::pyProng1);
+DECLARE_SOA_EXPRESSION_COLUMN(D0Pz, d0pz, float, 1.f * aod::hf_cand::pzProng0 + 1.f * aod::hf_cand::pzProng1);
+DECLARE_SOA_DYNAMIC_COLUMN(D0ImpactParameterProduct, d0impactParameterProduct,
+                           [](float dca1, float dca2) -> float { return dca1 * dca2; });
+DECLARE_SOA_DYNAMIC_COLUMN(D0M, d0m,
+                           [](float px0, float py0, float pz0, float px1, float py1, float pz1, const std::array<double, 2>& m) -> float { return RecoDecay::m(std::array{std::array{px0, py0, pz0}, std::array{px1, py1, pz1}}, m); });
+DECLARE_SOA_DYNAMIC_COLUMN(D0M2, d0m2, //!
```

Comment:
Please implement columns `invMassD0` and `invMassD0bar` with masses provided inside the definition instead of as arguments.